plugins/metric_links/src/Metric.php

```diff
@@ -243,41 +230,18 @@ public function isError(LinkStatus $status)
 
     /**
      * Get the machine name for a status
-     * 
-     * @param LinkStatus $status
+     *
+     * @param Page\Link $link
+     * @internal param Page\Link $status
      * @return string
      */
-    public function getMachineNameForStatus(LinkStatus $status)
-    {
-        if ($status->curl_code && empty($status->http_code)) {
-            return 'link_connection_error_' . $status->curl_code;
-        }
-        
-        return 'link_http_code_' . $status->http_code;
-    }
-
-    /**
-     * Get the links for a page
-     * 
-     * @param string $uri the uri of the page to get links for
-     * @param \DOMXPath $xpath the xpath of the page
-     * @return array an array of links
-     */
-    public function getLinks($uri, \DOMXPath $xpath)
+    public function getMachineNameForStatus(Page\Link $link)
     {
-        $links = \Spider::getUris(\Spider::getUriBase($uri), $uri, $xpath);
-
-        //Filter the links
-        foreach ($this->options['filters'] as $filter_class) {
-            $links = new $filter_class($links);
+        if ($link->original_curl_code && empty($link->original_status_code)) {
```

Comment:
This boolean condition is repeated a few times. Perhaps this is good candidate to add to the `Link` model.